### ISSUE TITLE
[Feature] Offset for bar tyyp

### ISSUE THREAD

**Issue by Andrea-LaRoss** [apache/echarts#20409]
### What problem does this feature solve?

Hi, In my project i have a timeseries that some line types series and some bars type series. One of the lines can be either a step start or a step end. Is it possible to align the bars type series to the lines maybe adding some type of offset?

### What does the proposed API look like?

![graphics_bars](https://github.com/user-attachments/assets/a3190984-e0ae-4997-ab8c-b4ad1ffa9546)

**Comment by helgasoft** [comment]
> align the bars type series to the lines

You didn't specify how to align... By default, line points are positioned in the **middle** of bars (your example chart).
Points could also be aligned to start or end of bars. Adding an **empty** bar series before or after the stacked ones creates the offset.
[Demo](https://echarts.apache.org/examples/en/editor.html?code=PYBwLglsB2AEC8sDeAoWsDGwA2wBOAXLANoDk2EA5gBZgBG2ArgKakA0plezz07p0GKw7cAJqQC6bNLAAeAQVkQAzkVTp0ogIZgtRMgFkY7WKQAqLE6QDqzcW1NnqjKwDE8EKwGUd3xnwkZAF9pdABPRRU1EJllZg9mVRIZdHUNdDAwkGYiUjotPBNlXQwAa1zi9hSNbV19AEYAJgAGBxbW2HqAVg6ADg6Adg76-uGAZmapavR8vABxLRBc5oBSUmqY9LT0zOzc2aKS8tNK0PTYWr0SJo72h26-jon70fuJqfPYWYWl01X19KbDRIWC7HKmA6wIGpaYXHRXYiNAAsHRRqOawwAbFjbi02tiPucwbkKNBhLDisxfqRigUwACNEEZIEggBuIA).
![image](https://github.com/user-attachments/assets/b1dadce4-c53b-4303-a0fe-7d78f8955ecd)
📌 please close issue if problem solved.

**Comment by Andrea-LaRoss** [comment]
Yes sorry for not specifing. What i meant to request was some type of property of bars series that allow to have an horizontal offset positioning them before or after a point of line series regardless of the barStack or the barGap properties and the addition of an empty bar series, because if i add a barWidth bars will return in the middle of points. Is there a workaround to maintain this type of offset even when i add barWidth?

**Comment by helgasoft** [comment]
> Is there a workaround to maintain this type of offset even when i add barWidth?

did you try _barWidth_ on all seeries, even the empty one ? it works.

**Comment by Andrea-LaRoss** [comment]
Yes like this works thank you very much. Last thing I need is this to work when I set barGap to '-100%' to get overlapping bars like here [Overlapping bars](https://echarts.apache.org/examples/en/editor.html?code=PYBwLglsB2AEC8sDeAoWsDGwA2wBOAXLANoDk2EA5gBZgBG2ArgKakA0plezz07p0GKw7cAJqQC6bNLAAeAQVkQAzkVTp0ogIZgtRMgFkY7WKQAqLE6QDqzcW1NnqjKwDE8EKwGUd3xnwkZAF9pdABPRRU1EJllZg9mVRIZDQ0kWDAwkGYiUjotPFJYGNT1VIysnNN8wtDy7V19AEYAJgAGB3aO2CaAVm6ADm6Adm6mprGAZjapFI0agHEtEFyAWgm2gFITLFxCaqZWOZK0ufRM7Nya9jPYBr0SVu6uhz7B7unXidfp2fLYRbLNYbbYOXb4XJcHh8Y7SOZlVL3fQtAAs3TR6LaYwAbDjnu1Ori_uULlVyBBoMJbsowMwVqYaQUwDscBDTGJSMcZIEggBuIA). Do you know any way to achieve that type of offset?

**Comment by helgasoft** [comment]
> get overlapping bars.. Do you know any way to achieve that type of offset?

yes, with two X-axes, [Demo](https://echarts.apache.org/examples/en/editor.html?code=CYQwLiC8AEDaDkBZA9gO3gGmvAKgVwFNNsB1A4Y3ACz0oDEAnAS0oGVw290BdAbgChkABzBM00GAG9-0aAGNkAG2QMAXHHiKmAcypgARosKZ42hgQLoM8VGiLXzFbhhnQAHgEE3TAM7rYrrKS0KAQ0AC-LrJBIeAgWD5UyADuqgBmIIo-BBFR0M6uAJ5evuqSka7ZzAR-cIHQ0tHRYIVCBOrw-iAMxKggALbtnQCMmPWyoSD-wwBMAAxY8wvQwwCsywAcywDsy8PDewDMcwVNsl0MAOIgQh1zAKTECspq2IbG9RVNwS1tHRfwXL1eqNM6_dpvbrEABafUGqhGxE83h8AElUMACG5VMM8k1JtM5nt5lh1lgtlhjlh9nsiaczhdrrdsA8nkoVB0zBZ0J88Q1oOD_lCsMjfOjMdjcUDvuNYhB_DMACzLZUqonUgBsey1ixJ0BmWvpTUF2C0qHssp8YAIzPgVu6YDZLw6jngn1c3H44V4QA)
![image](https://github.com/user-attachments/assets/ec42798a-cc3f-4315-a594-43d5018794d7)

**Comment by dberardo-com** [comment]
would be nice to be able to control the bar width to the left of its middle point and to the right, so that one could do something like barWidth: [0,100%] to have it all to the right of the middle point and [-100%,0] for the other behavior.

this would be similar to using a areaStyle: {} on a line series but still having access to the bar series API
